openshift/templates/api/api-deploy.yaml

```diff
@@ -110,9 +113,15 @@ objects:
       name: ${NAME}${SUFFIX}
       annotations:
         description: Defines how to deploy the application server
+        app.openshift.io/connects-to: >-
+          [{"apiVersion":"apps.openshift.io/v1","kind":"DeploymentConfig","name":"${AGENT_SERVICE_NAME}${SUFFIX}"},
+          {"apiVersion":"apps.openshift.io/v1","kind":"DeploymentConfig","name":"${DATABASE_DEPLOYMENT_NAME}${SUFFIX}"},
+          {"apiVersion":"apps.openshift.io/v1","kind":"DeploymentConfig","name":"${MSG_QUEUE_DEPLOYMENT_NAME}${SUFFIX}"},
+          {"apiVersion":"apps.openshift.io/v1","kind":"DeploymentConfig","name":"${SOLR_SERVICE_NAME}${SUFFIX}"}]
```

Comment:
Just wondering whether we should keep the naming consistent: other templates use `*_HOST` rather than `*_DEPLOYMENT_NAME`.